### ISSUE TITLE
Whether 3dMamba can apply DAC

### ISSUE THREAD

**Issue by NetfriendT** [gorkemcanates/Dual-Cross-Attention#14]
Thank you for inspiring me with your paper！The model I am currently using uses mamba module. A class U-net that uses mamba, can I apply your DAC module？I would like to ask whether the DAC module proposed in your paper can be modified and applied to the 3d segmentation model?I look forward to your answer and thank you

**Comment by gorkemcanates** [comment]
Yes, it can. You just need to modify the convolutions from 2D to 3D. Keep in mind that 3d cross-attention will increase the complexity of matrix multiplications. Make sure it does not exceed gpu memory.

**Comment by NetfriendT** [comment]
Thank you for your answer. However, I have encountered some problems at present, and I have replaced the 3d convolution according to your idea. However, I called DAC in the following way, and the error was reported as follows. Could you please give me some guidance? Thank you! @gorkemcanates 

The shapes of x1,x2,x3, and x4 are respectively [2,48,64,64,64][2, 96, 32, 32][2, 192, 16, 16][2, 384, 8, 8, 8], and it seems that an unexpected change in shape after passing into the DAC has caused the error.
![1](https://github.com/user-attachments/assets/7204f61f-4670-4ae9-9b59-eb1f048246e4)
![2](https://github.com/user-attachments/assets/925a74e2-54eb-455f-91d7-8f7010b7d0a2)
![3](https://github.com/user-attachments/assets/2368f2eb-4441-43e7-989a-c97f0f6cba4a)
